### PR TITLE
ui: fix drag and drop in FlatList while in strict mode

### PR DESCRIPTION
**Pull request by Forfold**
This PR fixes part of #2412 the drag and drop functionality when in Strict Mode by replacing the current DND library we use (`react-beautiful-dnd`) with `react-dnd`, as `react-beautiful-dnd` is no longer being maintained.

Also does some cleaning up in `FlatList`, simplifying some function calls and organizing the functions to make a bit more sense overall.

out of scope:
- removing drag and drop from `PolicyStepsCard`
- removing `react-beautiful-dnd` and `@types/react-beautiful-dnd` packages afterwards

### CODE REVIEW

**Review by tony-tvu — APPROVED** [review]
Functionally tested :white_check_mark:

**Review by tony-tvu — APPROVED** [review]

File: web/src/app/lists/DraggableListItem.tsx [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+  const style = {
+    transform: CSS.Transform.toString(transform),
+    transition,
+    backgroundColor: isDragging ? theme.palette.background.default : 'inherit',
+    cursor: 'pointer',
+    zIndex: isDragging ? 9001 : 1,
+  }
+
+  return (
+    <div ref={setNodeRef} style={style} {...attributes} {...listeners}>
```
Comment (KatieMSB) [inline web/src/app/lists/DraggableListItem.tsx:85]: Looks like the listeners here are blocking the event for the other actions button for the rotation users list.

**Review by KatieMSB — CHANGES_REQUESTED** [review]

**Comment by Forfold** [comment]
group review: reverse the states as in always show edit controls, toggle button hides them, and then shows drag handles. move edit toggle to the left/in line with avatars

**Comment by Forfold** [comment]
merged latest from master and wrapped up flipping the state of the toggle from the outcome of the group review, so this should be good to go again! @KatieMSB @tony-tvu

**Comment by Forfold** [comment]
good catch @tony-tvu! i think that's a focus issue- once focused on the list item you should be able to tab again to focus on the drag handle. when i migrated it to FlatList i forgot about that

**Review by tony-tvu — APPROVED** [review]
lgtm!

**Comment by Forfold** [comment]
whoops, needed to update the tests too @tony-tvu my bad! should be good to go now. running a full make check on my end to make sure the rest passes

File: web/src/app/lists/FlatList.tsx [hunk 2]
```diff
@@ -154,23 +133,58 @@ export default function FlatList({
...
+  const sensors = useSensors(
+    useSensor(PointerSensor),
+    useSensor(KeyboardSensor, {
+      coordinateGetter: sortableKeyboardCoordinates,
+    }),
+  )
+  const [dndItems, setDndItems] = useState(
+    // use IDs to sort, fallback to index
+    items.map((i, idx) => (i.id ? i.id : idx.toString())),
+  )
```
Comment (KatieMSB) [inline web/src/app/lists/FlatList.tsx:151]:
```suggestion
  const [dndItems, setDndItems] = useState(
    // use IDs to sort, fallback to index
    items.map((i, idx) => (i.id ? i.id : idx.toString())),
  )
  
  useEffect(() => {
    setDndItems(items.map((i, idx) => (i.id ? i.id : idx.toString())))
  }, [items])
```

The `dndItems` can get out of sync with the list `items` if users are removed from the rotation, adding a useEffect to keep updating on change seems to fix it

**Review by KatieMSB — CHANGES_REQUESTED** [review]

**Review by KatieMSB — APPROVED** [review]

**Review by tony-tvu — APPROVED** [review]

**Pull request merged**